PocketKit/Sources/PocketKit/Services.swift

```diff
@@ -216,6 +216,8 @@ struct Services {
         recentSavesWidgetUpdateService = RecentSavesWidgetUpdateService(store: UserDefaultsItemWidgetsStore(userDefaults: userDefaults, key: .recentSavesWidget))
         editorsPicksWidgetUpdateService = RecommendationsWidgetUpdateService(store: UserDefaultsItemWidgetsStore(userDefaults: userDefaults, key: .recommendationsWidget))
         widgetsSessionService = UserDefaultsWidgetSessionService(defaults: userDefaults)
+
+        tracker.track(event: Events.Widget.widgetCountEngagement(details: widgetsSessionService.widgetsDetails))
```

Comment:
as you already mention, this might not be the most appropriate place to track this event, because it only happens at app launch and has no context on widgets being installed and removed. I wonder if we can add a similar call in `getTimeline`, with the caveat that we need to filter those events. For example, we could store that dictionary somewhere, and then compare it with the dictionary we get by calling `getCurrentConfigurations` and, if there is a difference in widgets installed, we could fire an event of widgets installed/removed?

Reply:
I added the call in `getTimeline`, while it seems to work fairly well for installing a widget, it doesn't seem to trigger for uninstalling a widget. However, there will be an engagement call eventually once the code gets triggered again through some other actions.